graphene_sqlalchemy/types.py

```diff
@@ -21,70 +24,91 @@
 from .utils import get_query, is_mapped_class, is_mapped_instance
 
 
+class ORMField(OrderedType):
+    def __init__(
+        self,
+        type=None,
+        prop_name=None,
+        description=None,
+        deprecation_reason=None,
+        required=None,
+        _creation_counter=None,
+        **field_kwargs
+    ):
+        super(ORMField, self).__init__(_creation_counter=_creation_counter)
+        # The is only useful for documentation and auto-completion
+        common_kwargs = {
+           'type': type,
+           'prop_name': prop_name,
+           'description': description,
+           'deprecation_reason': deprecation_reason,
+           'required': required,
+        }
+        common_kwargs = {kwarg: value for kwarg, value in common_kwargs.items() if value is not None}
+        self.kwargs = field_kwargs
+        self.kwargs.update(common_kwargs)
+
+
 def construct_fields(
     obj_type, model, registry, only_fields, exclude_fields, connection_field_factory
 ):
     inspected_model = sqlalchemyinspect(model)
+    all_model_props = OrderedDict(
+        inspected_model.column_attrs.items() +
+        inspected_model.composites.items() +
+        [(name, item) for name, item in inspected_model.all_orm_descriptors.items()
+            if isinstance(item, hybrid_property)] +
+        inspected_model.relationships.items()
+    )
+
+    auto_orm_field_names = []
+    for prop_name, prop in all_model_props.items():
+        if (only_fields and prop_name not in only_fields) or (prop_name in exclude_fields):
+            continue
+        auto_orm_field_names.append(prop_name)
+
+    # TODO Get ORMField fields defined on parent classes
+    custom_orm_fields_items = []
+    for attname, value in list(obj_type.__dict__.items()):
```

Comment:
`.items()` is already a list; the `list()` is unnecessary. I'd consider using `iteritems()`, though. You could also build this up as a comprehension:

```
custom_orm_fields_items = [
    attname, value
    for attname, value in obj_type.__dict__.iteritems()
    if isinstance(value, ORMField)
]
```